### PR TITLE
Add check for result codes from HMI IsReady response

### LINKED ISSUE #1384: [SDLOPEN] SDL doesn't check result codes of HMI IsReady response
## Description:
SDL doesn't check result codes of HMI IsReady response

## Preconditions
SDL and HMI are started.

## Steps to reproduce
At InitHMI_OnReady send IsReady response with (<<error_result_code>>, available = true)

## Actual result
SDL accepts only available = true and interface is avalable, SDL doesn't check error_result_code

## Expected result
SDL should accept interface if is NOT available

## Environment
SDL repo: https://github.com/smartdevicelink/sdl_core/tree/feature/extended_policy_functionality
SDL commit: 7bd5326e8a6f5127220bf938a5be3063bdbb4ab4
OS with SDL: Ubuntu 14.04
VM used for OS with SDL (yes/no): no
ATF version: aa5beae6ab1726a9b43ed2fa15f66c20d7f7883f
Transport: WiFi

## Attachments
Problem can be reproduced with scripts:
https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/feature/extended_policy_functionality/test_scripts/API/IsReady/UI_IsReady/ATF_UI_IsReady_available_false_RAI.lua
https://github.com/CustomSDL/sdl_atf_test_scripts/blob/OS_DRAFT/test_scripts/API/IsReady/VR_IsReady/ATF_VR_IsReady_available_false_RAI.lua
https://github.com/CustomSDL/sdl_atf_test_scripts/blob/OS_DRAFT/test_scripts/API/IsReady/VR_IsReady/ATF_VR_IsReady_available_false_SingleRPC.lua

[Uploading ATF_VR_IsReady_available_false_RAI.lua.zip…]()

## Links
Requirements:
[UI Interface] SDL behavior in case HMI does not respond to UI.IsReady_request
[RegisterAppInterface] SDL behavior in case HMI does NOT respond to IsReady request
[UI Interface] HMI does NOT respond to IsReady and mobile app sends RPC that must be splitted

Clarification: Does SDL consider case VR.IsReady response with errorCode is the same as case VR.IsReady is not responded

## Expected delivery
- [ ]  Source code updates
- [ ]  Code comments
- [ ]  UTs add/update
- [ ]  ATF tests add/update
- [ ]  Manual tests
- [ ]  Add/update CI plans/jobs
- [ ]  SDD updates
- [ ]  Guidelines update (sdl_core_guides)
- [ ]  Guidelines update (sdl_hmi_integration_guidelines)

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #1384 and #3205 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This pull request is intended to fix issue with incorrect determination of interface availability, when HMI sends IsReady response. Corresponding to requirements, message with any error result code and "available" = "true" is equal to "available" = "false", when interface is set as unavailable. That's why check of result codes from IsReady response is added to ChangeInterfaceState method, also minor changes are added to corresponding requests for processing RPC to align their responses to requirements.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/commands/request_to_hmi.cc [hunk 1]
```diff
@@ -50,13 +51,46 @@ bool ChangeInterfaceState(ApplicationManager& application_manager,
...
-    application_manager.hmi_interfaces().SetInterfaceState(interface,
-                                                           interface_state);
+
+    if (!is_available) {
+      application_manager.hmi_interfaces().SetInterfaceState(
+          interface, HmiInterfaces::STATE_NOT_AVAILABLE);
+      return false;
+    }
+
+    if (response_from_hmi[strings::params].keyExists(hmi_response::code)) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/request_to_hmi.cc:61]: extract code block of parsing error to lambda or function

Comment (ychernysheva) [reply src/components/application_manager/src/commands/request_to_hmi.cc]: Done in commit 617fdd9

File: src/components/application_manager/src/commands/request_to_hmi.cc [hunk 2]
```diff
@@ -45,6 +45,24 @@ bool CheckAvailabilityHMIInterfaces(ApplicationManager& application_manager,
   return HmiInterfaces::STATE_NOT_AVAILABLE != state;
 }
 
+bool IsResponseCodeSuccess(
+    const smart_objects::SmartObject& response_from_hmi) {
+  hmi_apis::Common_Result::eType response_code =
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/request_to_hmi.cc:50]: auto

Comment (ychernysheva) [reply src/components/application_manager/src/commands/request_to_hmi.cc]: Done in commit 2c4728f

**Comment by ychernysheva** [comment]
@LuxoftAKutsan Corresponding unit tests are added and updated in commit 7241feb

**Comment by ychernysheva** [comment]
@theresalech This PR is ready for review.
Thank you!

**Comment by theresalech** [comment]
Hi @ychernysheva, issues 1384 and 3205 are not currently planned for the Core 6.2 release, but we will work to include if time allows.

**Comment by ychernysheva** [comment]
@jacobkeeler All conflicts were resolved in 1ef1011 after merging of current develop branch.
Also I found that some successful result codes weren't added to initial fix, so I've added them in 976532d 

But due to unclear regression status on current develop branch (too many fails on 4c9f873) we cannot define regression status on branch with this fix. May I suggest to wait until develop becomes more stable to merge it and then check regression status?

**Pull request merged**